libafl/src/inputs/multi.rs

```diff
@@ -0,0 +1,91 @@
+//! Definitions for inputs which have multiple distinct subcomponents.
+//!
+//! Unfortunately, since both [`serde::de::Deserialize`] and [`Clone`] require [`Sized`], it is not
+//! possible to dynamically define a single input with dynamic typing. As such, [`MultipartInput`]
+//! requires that each subcomponent be the same subtype.
+
+use alloc::{string::String, vec::Vec};
+
+use libafl_bolts::Error;
+use serde::{Deserialize, Serialize};
+
+use crate::inputs::Input;
+
+/// An input composed of multiple parts. Use in situations where subcomponents are not necessarily
+/// related, or represent distinct parts of the input.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MultipartInput<I> {
+    parts: Vec<I>,
+    names: Vec<String>,
+}
+
+impl<I> MultipartInput<I> {
+    /// Create a new multipart input.
+    pub fn new() -> Self {
```

Comment:
problably implementing From<[(I, &str); N]> is a good idea so you can do something like `MultipartInput::from([(BytesInput::new(), "foo") , (BytesInput::new(), "bar")])`